api/client/service.proto

```diff
@@ -37,10 +39,57 @@ message ExecutionResponse {
   }
 }
 
+message RequestStreamRequest {
+  uint64 amount = 1;
+}
+
+message Condition {
+  enum ConditionType {
+    EQUALS = 0;
+    LESS_THEN = 1;
+    LESS_OR_EQUAL = 2;
+    GREATER_THEN = 3;
+    GREATER_OR_EQUAL = 4;
+  }
+
+  string name = 1;
+  ConditionType type = 2;
+}
+
+message Expectation {
+  enum DataType {
+    TEXT = 0;
+    INT = 1;
+  }
+  string name = 1;
+  string input_header = 2;
+  DataType input_header_data_type = 3;
+  Condition condition = 4;
+}
+
+message RequestStreamResponse {
+  // method/path/uri/authority/content_length can all be expressed in request headers.
+  // so technically they are redundant. In case of ambiguity headers may end up being duplicated.
+  // Note that as of today, the target upstream/cluster that will be used to replay traffic to
+  // fixes the destination address/port
+  google.protobuf.StringValue method = 1;
+  google.protobuf.StringValue uri = 2; // XXX(oschaaf): path?
+  google.protobuf.StringValue authority = 3;
+  envoy.api.v2.core.HeaderMap headers = 4;
+  google.protobuf.UInt32Value content_length = 5;
+
+  repeated Expectation expectations = 6;
```

Comment:
Why do we have these complicated expectation conditions? Why not just exact match criteria?

Reply:
Well, I thought it might be useful to be able to express more then just exact matches, so we can set up expectations to match ranges of response-codes, and specify upper and/or lower bounds on expected response sizes. Also, see the original replay format draft, which suggests that being able to express this might be desirable.

Reply:
I think we should add this as needed. I think in general, we should have a framework that is relatively deterministic. It will send some request and get back the same thing each time.

Reply:
As per Slack discussion, +1 on simplifying request-level expecations. 
I'll simplify and modify this PR to prepare it for code-level review.